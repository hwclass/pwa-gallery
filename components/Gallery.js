import Router from 'next/router'
import uuid from 'uuid'

import Layout from './Layout';
import Image from './Image';

// import { TITLE } from '../constants/app'

export default class extends React.Component {
  showPhoto(e, uri) {
    e.preventDefault()
    Router.push(`/?photoUri=${uri}`, `/photo?uri=${uri}`)
  }

  render() {
    const { images } = this.props;
    return (
      <Layout title="PWA Gallery">
        <div>
          {images.map((image) =>
              <a
                key={uuid.v4()}
                className='photoLink'
                href={`/photo?id=${image.uri}`}
                onClick={(e) => this.showPhoto(e, image.uri)}
              >
                <Image id={uuid.v4()} key={uuid.v4()} uri={image.uri}/>
              </a>
            )
          }
          <style jsx>{`
            display: flex;
            height: 100%;
            display: -webkit-box;
            display: -moz-box;
            display: -ms-flexbox;
            display: -webkit-flex;
            align-items: center;
            justify-content: center;
          `}</style>
        </div>
      </Layout>
    );
  }
}
