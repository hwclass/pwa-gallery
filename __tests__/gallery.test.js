/* eslint-env jest */

import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import Gallery from '../components/Gallery.js'

describe('Gallery', () => {
  it('should list images', () => {
    const images = [{
      "uri": "some-url/for-image-1.jpg",
      "set": "111"
    }, {
      "uri": "some-url/for-image-2.jpg",
      "set": "222"
    }, {
      "uri": "some-url/for-image-3.jpg",
      "set": "333"
    }, {
      "uri": "some-url/for-image-4.jpg",
      "set": "444"
    }, {
      "uri": "some-url/for-image-5.jpg",
      "set": "555"
    }]
    const title = 'Some title for gallery'
    const gallery = shallow(<Gallery images={images} title={title}/>)
    expect(gallery.find('.photoLink').length).toEqual(5)
  })
})

describe('Gallery', () => {
  it('should match with snapshot', () => {
    const images = [{
      "uri": "some-url/for-image-1.jpg",
      "set": "111"
    }, {
      "uri": "some-url/for-image-2.jpg",
      "set": "222"
    }, {
      "uri": "some-url/for-image-3.jpg",
      "set": "333"
    }, {
      "uri": "some-url/for-image-4.jpg",
      "set": "444"
    }, {
      "uri": "some-url/for-image-5.jpg",
      "set": "555"
    }]
    const title = 'Some title for gallery'
    const component = renderer.create(<Gallery images={images} title={title}/>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})