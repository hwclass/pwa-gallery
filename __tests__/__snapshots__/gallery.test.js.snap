// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery should match with snapshot 1`] = `
<div
  className="jsx-3488773145"
>
  <header
    className="jsx-1922746274"
  >
    Car Photos
  </header>
  <div
    className="jsx-2977150191 gallery"
  >
    <a
      className="jsx-2977150191 photoLink"
      href="/photo?id=some-url/for-image-1.jpg"
      onClick={[Function]}
    >
      <article>
        <picture
          className="jsx-3329984789 pic"
        >
          <source
            className="jsx-3329984789"
            media="(min-width: 36em)"
            sizes="33.3vw"
            srcSet="http://some-url/for-image-1.jpg_27.jpg 1024w, http://some-url/for-image-1.jpg_27.jpg 640w, http://some-url/for-image-1.jpg$_27.jpg 320w"
          />
          <source
            className="jsx-3329984789"
            srcSet="http://some-url/for-image-1.jpg_27.jpg 2x, http://some-url/for-image-1.jpg_2.jpg 1x"
          />
          <img
            className="jsx-3329984789"
            src="http://some-url/for-image-1.jpg_27.jpg"
          />
        </picture>
        <span
          className="jsx-2158314212 title"
        >
          Some title for gallery
        </span>
      </article>
    </a>
    <a
      className="jsx-2977150191 photoLink"
      href="/photo?id=some-url/for-image-2.jpg"
      onClick={[Function]}
    >
      <article>
        <picture
          className="jsx-3329984789 pic"
        >
          <source
            className="jsx-3329984789"
            media="(min-width: 36em)"
            sizes="33.3vw"
            srcSet="http://some-url/for-image-2.jpg_27.jpg 1024w, http://some-url/for-image-2.jpg_27.jpg 640w, http://some-url/for-image-2.jpg$_27.jpg 320w"
          />
          <source
            className="jsx-3329984789"
            srcSet="http://some-url/for-image-2.jpg_27.jpg 2x, http://some-url/for-image-2.jpg_2.jpg 1x"
          />
          <img
            className="jsx-3329984789"
            src="http://some-url/for-image-2.jpg_27.jpg"
          />
        </picture>
        <span
          className="jsx-2158314212 title"
        >
          Some title for gallery
        </span>
      </article>
    </a>
    <a
      className="jsx-2977150191 photoLink"
      href="/photo?id=some-url/for-image-3.jpg"
      onClick={[Function]}
    >
      <article>
        <picture
          className="jsx-3329984789 pic"
        >
          <source
            className="jsx-3329984789"
            media="(min-width: 36em)"
            sizes="33.3vw"
            srcSet="http://some-url/for-image-3.jpg_27.jpg 1024w, http://some-url/for-image-3.jpg_27.jpg 640w, http://some-url/for-image-3.jpg$_27.jpg 320w"
          />
          <source
            className="jsx-3329984789"
            srcSet="http://some-url/for-image-3.jpg_27.jpg 2x, http://some-url/for-image-3.jpg_2.jpg 1x"
          />
          <img
            className="jsx-3329984789"
            src="http://some-url/for-image-3.jpg_27.jpg"
          />
        </picture>
        <span
          className="jsx-2158314212 title"
        >
          Some title for gallery
        </span>
      </article>
    </a>
    <a
      className="jsx-2977150191 photoLink"
      href="/photo?id=some-url/for-image-4.jpg"
      onClick={[Function]}
    >
      <article>
        <picture
          className="jsx-3329984789 pic"
        >
          <source
            className="jsx-3329984789"
            media="(min-width: 36em)"
            sizes="33.3vw"
            srcSet="http://some-url/for-image-4.jpg_27.jpg 1024w, http://some-url/for-image-4.jpg_27.jpg 640w, http://some-url/for-image-4.jpg$_27.jpg 320w"
          />
          <source
            className="jsx-3329984789"
            srcSet="http://some-url/for-image-4.jpg_27.jpg 2x, http://some-url/for-image-4.jpg_2.jpg 1x"
          />
          <img
            className="jsx-3329984789"
            src="http://some-url/for-image-4.jpg_27.jpg"
          />
        </picture>
        <span
          className="jsx-2158314212 title"
        >
          Some title for gallery
        </span>
      </article>
    </a>
    <a
      className="jsx-2977150191 photoLink"
      href="/photo?id=some-url/for-image-5.jpg"
      onClick={[Function]}
    >
      <article>
        <picture
          className="jsx-3329984789 pic"
        >
          <source
            className="jsx-3329984789"
            media="(min-width: 36em)"
            sizes="33.3vw"
            srcSet="http://some-url/for-image-5.jpg_27.jpg 1024w, http://some-url/for-image-5.jpg_27.jpg 640w, http://some-url/for-image-5.jpg$_27.jpg 320w"
          />
          <source
            className="jsx-3329984789"
            srcSet="http://some-url/for-image-5.jpg_27.jpg 2x, http://some-url/for-image-5.jpg_2.jpg 1x"
          />
          <img
            className="jsx-3329984789"
            src="http://some-url/for-image-5.jpg_27.jpg"
          />
        </picture>
        <span
          className="jsx-2158314212 title"
        >
          Some title for gallery
        </span>
      </article>
    </a>
  </div>
</div>
`;
